frontend/src/tests/UserIdentityCard.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { it, describe, beforeEach, expect} from 'vitest';
+import { mount } from '@vue/test-utils';
+import PrimeVue from 'primevue/config';
+import type { VueWrapper } from '@vue/test-utils/dist/vueWrapper';
+import type { DOMWrapper } from '@vue/test-utils/dist/domWrapper';
+import UserIdentityCard from '@/components/grantaccess/UserIdentityCard.vue';
+import { fixJsdomCssErr } from '@/tests/common/fixJsdomCssErr';
+
+fixJsdomCssErr()
+
+describe('UserIdentityCard', () => {
+    let wrapper: VueWrapper;
+
+    let card: DOMWrapper<HTMLElement>;
+    let cardEl: HTMLDivElement;
+    let username: DOMWrapper<HTMLElement>;
+    let usernameEl: HTMLSpanElement;
+
+    const props = {
+        userIdentity: {
+            userId: 'userId',
+            found: true,
+            firstName: 'First Name',
+            lastName: 'Last Name',
+        },
+    };
+
+    const propsNotFound = {
+        userIdentity: {
+            userId: 'userId',
+            found: false,
+            firstName: null,
```

Comment:
in the not found case, I feel the return data might only have userId, and found false, no firstName and lastName, but better double check